test/apps/iodemo/run_io_demo.sh

```diff
@@ -222,9 +222,12 @@ parse_args()
 
 check_params()
 {
-	if [ ! -x "${iodemo_exe}" ]
+	if [ -z "${iodemo_exe}" ]
 	then
 		error "missing executable from command line"
+	elif [ ! -x "${iodemo_exe}" ]
```

Comment:
IMO -x implies -z as well and -z check is redundant

Reply:
the error message was not accurate